revises network portion of 2.5 porting guide

acozine opened the pull request:
##### SUMMARY
Adds more detail, including the persistent connection options to the Network section of the 2.5 Porting Guide.

Fixes #36777

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
- Network Documentation

##### ANSIBLE VERSION
2.5 and higher

Commit:
revises network portion of 2.5 porting guide

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.5.rst:
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
+   ansible_network_os=ios
+   ansible_password=cisco
+   ansible_user=cisco
 
```
Worth linking to network docs here again?

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.5.rst:
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
+------------------------------------------------------------------
+
+Ansible 2.5 introduces two persistent connection types, ``network_cli`` and ``netconf``. We recommend you use these whenever possible.
+Unless you need a ``local`` connection, update your playbooks to use ``network_cli`` or ``netconf`` and to 
```
Maybe clarify that `local` is needed when using eAPI/NX-API?

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.5.rst:
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
+Unless you need a ``local`` connection, update your playbooks to use ``network_cli`` or ``netconf`` and to 
+specify your connection variables with standard Ansible connection variables:
+
+OLD In Ansible 2.4
```
`**OLD** In Ansible 2.4`

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.5.rst:
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
+
+We recommend setting connection properties in inventory by group. As you update your playbooks and inventory files, you can easily make the change to ``become`` for privilege escalation (on platforms that support it). For more information, see the :ref:`using become with network modules<become-network>` guide and the :ref:`platform documentation<platform_options>`. To update the ``ios_command`` t ...
+
+NEW In Ansible 2.5
```
`**NEW** In Ansible 2.5`

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.5.rst:
```diff
@@ -199,12 +199,16 @@ Porting custom scripts
 
+Top-level connection arguments like ``username``, ``host``, and ``password`` are deprecated and will be removed in version 2.9.
+
+OLD In Ansible 2.4
```
`**OLD** In Ansible 2.4`

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.5.rst:
```diff
@@ -216,36 +220,84 @@ Change in deprecation notice of top-level connection arguments
+       username: admin
+       password: admin
 
+NEW In Ansible 2.5
```
`**NEW** In Ansible 2.5`

Commit:
address gundalow comments

acozine commented:
@gundalow I addressed your comments in b3ff9bb8f014a6f29f5e0d8f16f3405a45d361cf - let me know if there are more changes you'd like to see, more links, more emphasis, etc.

Commit:
describes the value of persistent connections

Commit:
slims down the deprecation warning section

gundalow commented on docs/docsite/rst/porting_guides/porting_guide_2.5.rst:
```diff
@@ -199,12 +199,16 @@ Porting custom scripts
 
-Networking
-==========
+Network
```
I think somewhere prominent in this section we should link to the network guide.

Commit:
adds a link to the network docs landing page

acozine commented:
Chatted with Scott today - he says "if it looks good to you I'm fine with it." I'm going to take that as a positive review and merge.

gundalow commented:
In `stable-2.5` so marking as done
